### PR TITLE
MNT Fix all remaining estimator checks (except pickle related errors)

### PR DESCRIPTION
**Pull request by TamaraAtanasoska**
## Description
This PR fixes all the remaining `scikit-learn` estimator checks, excluding the pickle errors that will come in a subsequent PR by itself. 

@adrinjalali check it out! I will add some comments where I have doubts. 
## Tests

- [ ] no new tests required
- [ ] new tests added
- [x] existing tests adjusted

## Documentation

- [ ] no documentation changes needed
- [ ] user guide added or updated
- [x] API docs added or updated
- [ ] example notebook added or updated

### CODE REVIEW

File: test/unit/adversarial/helper.py [hunk 1]
```diff
@@ -232,11 +235,11 @@ def generate_data_combinations(n=10):
...
         assert X + Y * K + Z * K * K == c_orig
         X_type = dist_type[X]
-        X = datas[X]
+        X = dataX[X]
         Y_type = dist_type[Y]
-        Y = datas[Y]
+        Y = dataY[Y]
         Z_type = dist_type[Z]
-        Z = datas[Z]
+        Z = dataY[Z]
```
Comment (TamaraAtanasoska) [inline test/unit/adversarial/helper.py:242]: @adrinjalali I tried to change as little as possible here and maintain consistency with the estimator checks changes - does this make sense to you? Y is required to be one dimensional according to the estimator checks so I have added real one dimensional options, added the same dimensionality to Z.

**Comment by TamaraAtanasoska** [comment]
@adrinjalali this is reviewable, the test is fixed now. check the comment above pls too!

**Review by adrinjalali — COMMENTED** [review]
Otherwise LGTM.

File: fairlearn/adversarial/_adversarial_mitigation.py [hunk 2]
```diff
@@ -412,7 +414,9 @@ def fit(self, X, y, *, sensitive_features=None):
             Array-like containing the sensitive features of the
             training data.
         """
-        X, y, A = self._validate_input(X, y, sensitive_features, reinitialize=True)
+        first_call = not hasattr(self, "classes")
```
[the remark this answers is not in the document]
Comment (TamaraAtanasoska) [reply fairlearn/adversarial/_adversarial_mitigation.py:417]: Oops glad you caught this 😬

File: fairlearn/adversarial/_adversarial_mitigation.py [hunk 3]
```diff
@@ -412,7 +414,9 @@ def fit(self, X, y, *, sensitive_features=None):
...
+
+        X, y, A = self._validate_input(X, y, sensitive_features, first_call)
```
[the remark this answers is not in the document]
Comment (adrinjalali) [reply fairlearn/adversarial/_adversarial_mitigation.py:419]: yeah it just needs to be done before the 1.6 sklearn release

**Comment by TamaraAtanasoska** [comment]
> test failing

passed, repeated OpenML failures :(

**Review by adrinjalali — APPROVED** [review]

**Pull request merged**